debian/rules

```diff
@@ -276,46 +64,57 @@ override_dh_auto_install:
 	cp /usr/share/fonts/truetype/ubuntu/Ubuntu-R.ttf \
 		debian/tmp/usr/share/fonts/Plymouth.ttf
 
-	dh_auto_install --destdir=debian/tmp --buildsystem=meson+ninja --sourcedirectory=vendor/systemd
-	for p in $(CLEAN_PATHS); do		\
-		rm -r debian/tmp/$$p ;		\
-	done
 	touch debian/tmp/etc/machine-id
+
+	# Include all needed systemd bits
+	set -e;				\
+	for f in $$(dpkg -L systemd |	\
+	grep -E '(^/lib/systemd/system|modprobe\.d/|sysctl\.d/|rules\.d/|tmpfiles\.d/|bin/)')	\
+	$$(dpkg -L systemd-sysv | grep sbin/)			\
+	/var/lib/systemd/					\
+	$$(dpkg -L udev | grep -E '(bin/|lib/|rules\.d/)')	\
+		;						\
+	do							\
+		LD_PRELOAD=					\
+			/usr/lib/dracut/dracut-install		\
+			-D $(CURDIR)/debian/tmp --ldd $$f;	\
+	done
+	# But we don't won't to automount the rootfs
+	rm debian/tmp/usr/lib/systemd/system-generators/systemd-gpt-auto-generator
+	# or this (binfmt_misc module is not in the initramfs)
+	rm debian/tmp/usr/lib/systemd/system/sysinit.target.wants/proc-sys-fs-binfmt_misc.automount
+
 	# This hack should be removed with #113
 	sed -i '/^After=/{;s, *plymouth-start[.]service *, ,;/^After= *$$/d;}' debian/tmp/usr/lib/systemd/system/systemd-ask-password-*
 
-	LD_LIBRARY_PATH=`pwd`/debian/tmp/usr/lib/*/:`pwd`/debian/tmp/usr/lib/systemd `pwd`/debian/tmp/usr/bin/systemd-hwdb --root debian/tmp update --usr --strict
+	# XXX Is this still needed?
```

Comment:
I do not find the reason in the history. Maybe the best would be to remove it and try. We can also do that in a follow up PR to make it easy to revert.

Reply:
systemd udev requires hwdb database, one can ship raw files of it, or just the compiled binary form (optimized), or both.
On .deb based systems we ship hwdb source files, and compile the database in maintainer scripts (such that one can optionally install systemd-hwe, or 3rd party or personal hwdb files and still have optimal boot).

Inside the initrd however, it is preffered for speed and size to compile hwdb sources files and only ship the binary database alone. See hwdb and systemd-hwdb manpages.

Reply:
If it is not clear what this is doing, then please write a descriptive comment based on the above. If you drop generating a hwdb the initrd will become slower and bigger and @kubiko will haunt you.

Reply:
It is not about dropping hwdb. The question is why we need to move it from `/usr/lib/systemd` to `/usr/bin`?

Reply:
I think I misread it...